oidc-api/src/main/java/uk/gov/di/authentication/oidc/validators/RequestObjectAuthorizeValidator.java

```diff
@@ -214,10 +214,28 @@ private static boolean isSignatureValid(SignedJWT signedJWT, String publicKey) {
     private Optional<AuthRequestError> validateVtr(JWTClaimsSet jwtClaimsSet, URI redirectURI) {
         List<String> authRequestVtr = new ArrayList<>();
         try {
-            authRequestVtr =
-                    Objects.isNull(jwtClaimsSet.getClaim(VTR_PARAM))
-                            ? emptyList()
-                            : List.of(jwtClaimsSet.getClaim(VTR_PARAM).toString());
+            if (Objects.nonNull(jwtClaimsSet.getClaim("vtr"))) {
+                if (jwtClaimsSet.getClaim("vtr") instanceof String vtr) {
+                    authRequestVtr = List.of(vtr);
+                } else if (jwtClaimsSet.getClaim("vtr") instanceof List<?> vtrList
+                        && vtrList.stream().allMatch(vtr -> vtr instanceof String)) {
+                    authRequestVtr =
+                            List.of(
+                                    String.format(
+                                            "[\"%s\"]",
+                                            String.join(
+                                                    "\",\"",
+                                                    jwtClaimsSet.getStringArrayClaim("vtr"))));
+                } else {
+                    LOG.error("vtr in AuthRequest is not valid. Could not be parsed");
+                    return errorResponse(
+                            redirectURI,
+                            new ErrorObject(
+                                    OAuth2Error.INVALID_REQUEST_CODE, "Request vtr not valid"));
+                }
+            } else {
+                authRequestVtr = emptyList();
+            }
             var vectorOfTrust = VectorOfTrust.parseFromAuthRequestAttribute(authRequestVtr);
```

Comment:
I think this is the only call that throws `IllegalArgumentException` so the `try` block could be moved down to here

Reply:
In that case we'd need two try blocks for the two different exception blocks - not sure if that's preferable or not 🤷